### ISSUE TITLE
Venues add images when creating their listing

### ISSUE THREAD

**Issue by ctrembath** [club-soda/club-soda-guide#58]
As a venue creating my venue listing,
I'd like to add up-to 4 images, 
so mindful drinkers know what my venue looks like

## Acceptance Criteria
+ [x] When I look at my venue page I see an 'Upload photo' button
+ [x] Clicking on this button takes me to a form where I can upload my photo
+ [x] I can upload a venue photo successfully
+ [x] I have some form of feedback to let me know that an upload is in progress
+ [x] If my upload fails I get a user-friendly error message
+ [x] If my upload is successful, I am taken back to my venue page
  + [x] Photos is displayed across the top of the venue page, centred
  + [x] If a photo already existed, substitute the new photo for the old

Stretch goals
+ [x] Venues can upload up to 4 photos
+ [x] Photos are displayed as a 'carousel' in venue page #28

**Comment by iteles** [comment]
@Cleop @Danwhy Let's go for one image first and test with users!

**Comment by Cleop** [comment]
@jussi-club-soda - do you have an existing AWS account? 

We'd like to use AWS S3 for hosting the uploaded images from venues etc. and so it'd be useful to know if you already have an account in use.

**Comment by jussi-club-soda** [comment]
@Cleop We do have an AWS account, which seems to be "suspended" at the moment. We got free credits from somewhere a few years ago, and planned to use it for one of our websites, but then decided it was just too complex so never installed anything there in the end. I could get it reinstated, or close it down and open a new one. We could probably get free credits again from somewhere too :)

**Comment by nelsonic** [comment]
@jussi-club-soda if you are not using your existing account for anything, I would _recommend_ you open a new account to take advantage of the "free tier" for 12 months. 
(_this will "save" you around $20 per year ... so if you prefer to reinstate the existing account go for it!_)  
It does not need to be done right now, we will remind you when it's needed. 👍

**Comment by Cleop** [comment]
Image upload is now working from phoenix to S3 on my dwyl.com account. We can update this to be a CS account once this has been decided on.

**Comment by Cleop** [comment]
I'm trying to create the conditional behaviour for if an image exists or not in order to display the default 'add image' background image when an image hasn't been uploaded yet.

I've tried implementing the solution in this SO article:
https://stackoverflow.com/questions/980855/inputting-a-default-image-in-case-the-src-attribute-of-an-html-img-is-not-vali

When you provide a real 'type' like ‘image/jpeg’ then you get the img loading when the img exists but you get an error when it doesn't exist rather than the backup image. If you give a type that doesn't exist then the back up image loads but the real image doesn’t load. If you remove ‘type’ (the docs say 'type' is optional so in theory it should work without it) the back up image still doesn’t work and when you do have a valid image it tries to download it onto your computer rather than displaying it.

This is the error message when you have 'type="image/jpeg" and give an image that doesn't exist:
![image](https://user-images.githubusercontent.com/16775804/47848259-a917b880-ddc5-11e8-9f8b-babbd94442cf.png)
Rather than showing the back up image.

I'm looking into an alternative method of doing this or whether I can understand that error message any better.

**Comment by Cleop** [comment]
See: https://hexdocs.pm/ex_aws_s3/ExAws.S3.html#module-more-high-level-functionality for uploading multiple images at a time

**Comment by Cleop** [comment]
@jussi-club-soda @nelsonic - Club Soda is currently using an account I created with my own details which qualified but is exceeding the free tier:

![image](https://user-images.githubusercontent.com/16775804/51481382-a8227180-1d8b-11e9-9c26-04247bf99021.png)

@jussi-club-soda please can you provide us with the details to access your previous account or let us know if you'd like us to set up a new one for you (and provide what email/ payment card etc. to link that to)? Then we can transfer the existing data from my account.

**Comment by nelsonic** [comment]
@Cleop _first_ we need to migrate this content to an S3 account controlled by Club Soda ASAP.
_second_ we need to understand why `DataTransfer` costs are so high ... 🤔 
Please check the average size of object in the S3 bucket and confirm which is the most frequently downloaded item (_if you can_).

**Comment by Cleop** [comment]
Agreed during demo that @jussi-club-soda would set up a CS AWS account and share the details with us once this is done so that we can transfer the data to that account.

**Comment by Cleop** [comment]
I've written the script for making the submit button say 'Uploading...' and I'm trying to add it to the appropriate files. I've tried the following:

```
<script src="<%= static_path(@conn, "/js/photo-uploading.js") %>"></script>
```
```
<script type="text/javascript" src="/js/photo-uploading.js"></script>
```
It works when I add it to `app.js` like this but then that means it's being run on every page (which it doesn't need to and so then there's an error because the script is looking for an id that doesn't exist):
```
require("./photo-uploading.js")
```

I've tried changing the file path by adding assets to it but with no luck. This is the error message I get:
<img src="https://user-images.githubusercontent.com/16775804/52112982-fa6f5800-25ff-11e9-9ddc-5153820aa296.png" width=400px />

Any suggestions @RobStallion @SimonLab @Danwhy ?

**Comment by Danwhy** [comment]
@Cleop The only js file we're actually serving is the app.js. The error you're getting is because that file leads to a 404 page which is html, not js.

We'd have to either change the brunch config to serve that file, or add the code to the app.js in an if statement so it only runs if the button element exists
